lightning/src/util/config.rs

```diff
@@ -59,6 +59,14 @@ pub struct ChannelHandshakeConfig {
 	/// Default value: 1. If the value is less than 1, it is ignored and set to 1, as is required
 	/// by the protocol.
 	pub our_htlc_minimum_msat: u64,
+	/// Set to the largest value HTLC we will accept to process.
+	///
+	/// If configured, this value is sent to our counterparty on channel-open and we close the channel any time
+	/// our counterparty misbehaves by sending us an HTLC with a value larger than this.
```

Comment:
I think other impls just fail the HTLC rather than close on the counterparty -- which I'd prefer as well and I believe the spec doesn't require closing on violation.

Reply:
Same being discussed [here](https://github.com/rust-bitcoin/rust-lightning/pull/651#discussion_r449231788): whether to make it configurable and whether to enforce it.